oceanmodes/test/test_modes.py

```diff
@@ -100,3 +106,152 @@ def test_N2_const_equal_spacing(self):
+        # check for unequal lengths of arrays
+        with self.assertRaises(ValueError):
+            _, _, _, _, _, _ = modes.instability_analysis_from_N2_profile(
+                depth[1:], N2, 1., beta, 20, 20, 1e-1, 1e-1, ubar, vbar, etax, etay
+            )
+        with self.assertRaises(ValueError):
+            _, _, _, _, _, _ = modes.instability_analysis_from_N2_profile(
+                depth, N2[1:], 1., beta, 20, 20, 1e-1, 1e-1, ubar, vbar, etax, etay
+            )
+        #with self.assertRaises(ValueError):
+        #    _, _, _, _, _, _, _, _ = modes.instability_analysis_from_N2_profile(
+        #        depth, N2, 1., beta, 50, 50, 1e-1, 1e-1, ubar[1:], vbar, etax, etay
+        #    )
+
+        depth_non_monotonic = depth
+        depth_non_monotonic[0] = 5
+        # check for non-monotonic profile
+        with self.assertRaises(ValueError) as cm:
+            _, _, _, _, _, _ = modes.instability_analysis_from_N2_profile(
+                depth_non_monotonic, N2, 1., beta, 50, 50, 1e-1, 1e-1, ubar, vbar, etax, etay
+            )
+
+    def test_N2_const_equal_spacing(self):
+        # prepare a test N^2 profile
+        nz = 20
+        zin = 0.5*nz**-1 + np.arange(nz, dtype=np.float64)/nz
+        N2 = np.linspace(1e-3, 1., nz)
+        # N2 = np.full(nz, 1.)
+        f0 = 1.
+        beta = 1e-6
+        # beta = 0.
+        Nx = 20
+        Ny = 20
+        dx = .1
+        dy = .1
+        # ubar = np.zeros((nz+1, 1))
+        # vbar = np.zeros((nz+1, 1))
+        # ubar = 1e-2 * np.ones((nz+1, 1))
+        # ubar[:nz/2, 0] = 1e-2 * np.arange(nz/2, 0., -1, dtype=np.float64)/nz*2
+        # ubar[nz/2:, 0] = 1e-2 * np.arange(nz/2+1, dtype=np.float64)/nz*2
+        ubar = 1e-2 * np.arange(nz+1)
+        vbar = 1e-2 * np.ones((nz+1, 1))
+        etax = np.zeros((2, 1))
+        etay = np.zeros((2, 1))
+
+        k, l, z, max_growth_rate, growth_rate, vertical_modes = modes.instability_analysis_from_N2_profile(
+                zin, N2, f0, beta, Nx, Ny, dx, dy, ubar, vbar, etax, etay, sort='LI', num=4
+        )
+        
+        # make sure we got the right number of modes
+        # NOPE! don't want all the modes and scipy.sparse.linalg.eigs won't
+        # compute them
+        # self.assertEqual(nz, len(def_radius))
+        # make sure the modes themselves have the right structure
+        self.assertEqual(nz+1, vertical_modes.shape[0],
+            msg='modes array must be in the right shape')
+
+        self.assertTrue(np.all( np.diff( growth_rate.reshape((growth_rate.shape[0], len(k)*len(l))).imag.max(axis=1) ) < 0.),
+        #self.assertTrue(np.all( max_growth_rate == 0.),
+            msg='imaginary part of modes should be descending')
+
+        #nmodes = len(def_radius)
+        #zero_crossings = np.abs(np.diff(np.sign(bc_modes), axis=0)/2).sum(axis=0)
+        #self.assertListEqual(list(range(nmodes)), list(zero_crossings),
+        #    msg='modes should have the correct number of zero crossings')
```

Comment:
Can you think of any tests that will actually verify whether the output is correct?